presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1917,6 +1918,26 @@ public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHa
         return handle.getTableParameters().isEmpty() || !isFullAcidTable(handle.getTableParameters().get());
     }
 
+    @Override
+    public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, Optional<String> schemaName)
+    {
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
+        for (SchemaTableName name : listViews(session, schemaName)) {
+            try {
+                getView(session, name).ifPresent(view -> views.put(name, view));
+            }
+            catch (PrestoException e) {
+                if (e.getErrorCode().equals(HIVE_VIEW_TRANSLATION_ERROR.toErrorCode())) {
```

Comment:
What's the implication of not adding failing views to the view map? What would happen if one of these views was referenced in a query?

Reply:
view resolution shouldn't be impacted, as getViews method is only used for listing views and columns.